components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests

```diff
@@ -141,21 +144,58 @@ function install_kfp() {
   echo "[Installing KFP] Pipeline pods are ready"
 }
 
-function generate_iam_role_name() {
+function generate_oidc_role_name() {
   OIDC_ROLE_NAME="$(echo "${DEPLOY_NAME}-kubeflow-role" | cut -c1-64)"
-  OIDC_ROLE_ARN="arn:aws:iam::$(aws sts get-caller-identity --query=Account --output=text):role/${OIDC_ROLE_NAME}"
+  OIDC_ROLE_ARN="arn:aws:iam::${AWS_ACCOUNT_ID}:role/${OIDC_ROLE_NAME}"
 }
 
-function install_generated_role() {
+function install_oidc_role() {
   kubectl patch serviceaccount -n ${KFP_NAMESPACE} ${KFP_SERVICE_ACCOUNT} --patch '{"metadata": {"annotations": {"eks.amazonaws.com/role-arn": "'"${OIDC_ROLE_ARN}"'"}}}'
 }
 
-function delete_generated_role() {
+function delete_oidc_role() {
   # Delete the role associated with the cluster thats being deleted
   aws iam detach-role-policy --role-name "${OIDC_ROLE_NAME}" --policy-arn arn:aws:iam::aws:policy/AmazonSageMakerFullAccess
+  aws iam delete-role-policy --role-name "${OIDC_ROLE_NAME}" --policy-name AllowAssumeRole
   aws iam delete-role --role-name "${OIDC_ROLE_NAME}"
 }
 
+function generate_assumed_role() {
```

Comment:
There was a discussion to add more integration tests - we should test that the assumed role works as expected, even if only within the same account.

Reply:
I have an integration test that creates a training job, and another that creates a processing job. Neither will work if there is an issue with the roles, or if there is an issue with our implementation.

Reply:
I may be missing this but could you elaborate - if the provided role does not exist, sure the request will fail. But how do I know that the permissions I am adding via this assumed role are actually available to the client ?

Reply:
What do you mean by "available to the client"?

Reply:
As discussed offline, there is no simple way of testing this, so ok for now.